libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/PlannedRouteRefreshController.kt

```diff
@@ -131,11 +134,12 @@ internal class PlannedRouteRefreshController @VisibleForTesting constructor(
         routeRefresherResult.fold(
             { logW("Planned route refresh error: $it", RouteRefreshLog.LOG_CATEGORY) },
             {
-                if (it.success) {
+                attemptListener.onRoutesRefreshAttemptFinished(it)
+                if (it.anySuccess()) {
                     stateHolder.onSuccess()
                     listener.onRoutesRefreshed(it)
                 } else {
-                    if (retryStrategy.shouldRetry()) {
+                    if (it.anyRequestFailed() && retryStrategy.shouldRetry()) {
```

Comment:
Do you feel like planned and immediate route refresh controller duplicate notification logic, which seems not so easy right now?

Reply:
Not really, the logic is different. Some things are common, but most of them aren't.
The logic in the line that you left a comment to is not duplicated, for example.